lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -745,18 +751,20 @@ def field_to_label(record, field, labels, label)
       record.delete(field)
     end
 
-    def set_payload(record, entry, is_container_json)
+    def set_payload(record, entry, is_json)
       # Use textPayload if
-      # 1. This is a Cloud Functions log that matched the expected regexp
-      # 2. This is a Cloud Functions log and the 'log' key is available
-      # 3. This is an unstructured Container log and the 'log' key is available
-      # 4. The only remaining key is 'message'
-      if @service_name == CLOUDFUNCTIONS_SERVICE && @cloudfunctions_log_match
+      # 1. The log was not valid JSON
```

Comment:
I think you mean for the comment to be:

# Use text payload if the log was not valid JSON and:
# 1. [...]
etc.

Since that's effectively being AND'ed with all the other optional conditions.